### ISSUE TITLE
Restart rackspace-monitoring agent if not responding

### ISSUE THREAD

**Issue by fernandohonig** [rackspace-cookbooks/rackspace_monitoring#32]
When chef-client runs, should be able to identify if the rackspace-monitoring-agent is not responding. If not responding, Chef should restart it.

**Comment by martinb3** [comment]
Is this when agent is hanging, or when agent is not running at all?

**Comment by fernandohonig** [comment]
We executed a stress test in a customer node and the agent got disconnected.

**Comment by jujugrrr** [comment]
It looks like something that should be managed by init and the OS/package rather than Chef. @fernandohonig / @martinb3 thoughts ?

> We executed a stress test in a customer node and the agent got disconnected.

@fernandohonig can you give more details ? Did the service stop ? Was the PID still there ? How did you fix it ?

**Comment by martinb3** [comment]
Yes, I agree @jujugrrr. I'm not even sure if this might be a feature -- do we know if the agent throttles itself under high load? And also, if this load caused any kind of network outage, or packet loss, I'm not sure restarting it would help.

More details could shed some light on possible courses of action :+1:

**Comment by rphillips** [comment]
The logs for the agent would be super helpful as well. They are located in `/var/log/rackspace-monitoring-agent.log`.
